pkg/connector/slack/v0/connector_test.go

```diff
@@ -0,0 +1,172 @@
+package slack
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/instill-ai/component/pkg/base"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	apiKey = "testkey"
+)
+
+func TestConnector_ExecuteWriteTask(t *testing.T) {
+	c := qt.New(t)
+	ctx := context.Background()
+	bc := base.Connector{Logger: zap.NewNop()}
+	connector := Init(bc)
+
+	testcases := []struct {
+		name     string
+		input    UserInputWriteTask
+		wantResp WriteTaskResp
+		wantErr  string
+	}{
+		{
+			name: "ok to write",
+			input: UserInputWriteTask{
+				ChannelName:     "test_channel",
+				Message:         "I am unit test",
+				IsPublicChannel: true,
+			},
+			wantResp: WriteTaskResp{
+				Result: "succeed",
+			},
+		},
+		{
+			name: "fail to write",
+			input: UserInputWriteTask{
+				ChannelName:     "test_channel_1",
+				Message:         "I am unit test",
+				IsPublicChannel: true,
+			},
+			wantErr: `there is no match name in slack channel \[test_channel_1\]`,
+		},
+	}
+
+	for _, tc := range testcases {
+		c.Run(tc.name, func(c *qt.C) {
+
+			connection, err := structpb.NewStruct(map[string]any{
+				"api_key": apiKey,
+			})
+			c.Assert(err, qt.IsNil)
+
+			// It will increase the modification range if we do DI to CreateExecution.
+			// So, I replaced it with the code below.
+			// exec, err := connector.CreateExecution(nil, connection, taskWriteMessage)
+			// c.Assert(err, qt.IsNil)
+			e := &execution{
+				ConnectorExecution: base.ConnectorExecution{Connector: connector, SystemVariables: nil, Connection: connection, Task: taskWriteMessage},
+				client:             &MockSlackClient{},
+			}
+			e.execute = e.sendMessage
+			exec := &base.ExecutionWrapper{Execution: e}
+
+			pbIn, err := base.ConvertToStructpb(tc.input)
+			c.Assert(err, qt.IsNil)
+
+			got, err := exec.Execution.Execute(ctx, []*structpb.Struct{pbIn})
+
+			if tc.wantErr != "" {
+				c.Assert(err, qt.ErrorMatches, tc.wantErr)
+				return
+			}
+
+			wantJSON, err := json.Marshal(tc.wantResp)
+			c.Assert(err, qt.IsNil)
+			c.Check(wantJSON, qt.JSONEquals, got[0].AsMap())
+		})
+	}
+}
+
+func TestConnector_ExecuteReadTask(t *testing.T) {
+
+	c := qt.New(t)
+	ctx := context.Background()
+	bc := base.Connector{Logger: zap.NewNop()}
+	connector := Init(bc)
+
+	testcases := []struct {
+		name     string
+		input    UserInputReadTask
+		wantResp ReadTaskResp
+		wantErr  string
+	}{
+		// {
```

Comment:
@donch1989 
The test failed on GitHub Action because the DateTime in ThreadReplyMessage is not the same.
There is a 1 hour difference and I guess it is the time-zone difference.
I will confirm how to modify it.
Maybe I will mock `func transformTSToDate` .

**Question**
Do you have the experience of dealing with it before?
How will you recommend to do the test for it?

Reply:
And, I also want to confirm if we can postpone it when we cannot make it for tomorrow release.

Reply:
try this one
```
timestamp.Format(time.RFC3339)) 
```

We have another release this Thursday. I think it is fine if we can't make it tomorrow.

Reply:
@donch1989 
It will still fail because time.Format locally will be different time.Format in GitHub Action.
Normally, we should mock this part.
But, I am not quite sure how much time it will take to change the structure to mock the code.

So, I will change the direction to skip checking Datetime in the test code first.
Will it be ok? What do you think about it?

btw, I changed `transformTSToDate(msg.Timestamp, "2006-01-02 15:04:05")` to `transformTSToDate(msg.Timestamp, time.RFC3339)`

Reply:
Checking if [this package](https://go.dev/src/time/time_test.go) can solve my problem

Reply:
Memo
Will keep trying tomorrow.
Before that, I make the test code pass in GitHub Action but failed in local.

Reply:
I just simply call the same function to make the result consistent in local and in GitHub Action.

Reply:
> I just simply call the same function to make the result consistent in local and in GitHub Action.
Not sure what you mean. I think we can just ensure it always uses the UTC+0 timezone.